Flaky unit tests

<!---
Hi!
Have you tried searching for your issue on the following forums?
If you have any questions, please ask them there.

Forums:
 - TLA⁺ Google Groups forum: https://groups.google.com/g/tlaplus/
 - GitHub Discussions forum: (TBA?)

Thanks!
-->

## Description
The unit tests sometimes fail in GitHub CI. Two examples:
 - https://github.com/tlaplus/tlaplus/pull/925#issuecomment-2115885840
 - CI failure for a commit that passed PR CI and local tests on my machine: https://github.com/tlaplus/tlaplus/actions/runs/9487536859/attempts/1 --- unfortunately the log does not include the test output :(

I don't know if the examples are related, but they may be pointing at some obscure race condition in TLC or in the test infrastructure.

<img width="1295" alt="image" src="https://github.com/tlaplus/tlaplus/assets/88777/381b8174-772e-48b4-82e0-9155732265e1">


```java
util.Assert$TLCRuntimeException: Unable not rename file during the clean-up.
	at util.Assert.fail(Assert.java:101)
	at tlc2.tool.fp.DiskFPSet$Flusher.mergeNewEntries(DiskFPSet.java:1114)
	at tlc2.tool.fp.DiskFPSet$Flusher.flushTable(DiskFPSet.java:1064)
	at tlc2.tool.fp.MSBDiskFPSetTest2.testHighFingerprint2(MSBDiskFPSetTest2.java:87)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
Caused by: java.nio.file.FileSystemException: C:\Users\RUNNER~1\AppData\Local\Temp\tlc2.tool.fp.MSBDiskFPSetTest21718993886316.fp: The process cannot access the file because it is being used by another process.

	at java.base/sun.nio.fs.WindowsException.translateToIOException(WindowsException.java:92)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:103)
	at java.base/sun.nio.fs.WindowsException.rethrowAsIOException(WindowsException.java:108)
	at java.base/sun.nio.fs.WindowsFileCopy.move(WindowsFileCopy.java:384)
	at java.base/sun.nio.fs.WindowsFileSystemProvider.move(WindowsFileSystemProvider.java:292)
	at java.base/java.nio.file.Files.move(Files.java:1422)
	at util.FileUtil.replaceFile(FileUtil.java:219)
	at tlc2.tool.fp.DiskFPSet$Flusher.mergeNewEntries(DiskFPSet.java:1112)
```

--- a/tlatools/org.lamport.tlatools/test/tlc2/TraceExpressionSpecTest.java
+++ b/tlatools/org.lamport.tlatools/test/tlc2/TraceExpressionSpecTest.java
@@ -37,20 +37,18 @@ import util.SimpleFilenameToStream;
 
 public abstract class TraceExpressionSpecTest extends ModelCheckerTestCase {
 
-	private static final String OUTPUT_PATH = "states"; // Rest of tests dump stuff here.
-
 	public TraceExpressionSpecTest(final String specName, final String configName, String mode, int exitStatus) {
 		this(specName, "TESpecTest", configName, mode, exitStatus);
 	}
 
 	public TraceExpressionSpecTest(final String specName, final String path, final String configName, String mode, int exitStatus) {
 		super(specName, path,
-				new String[] { "-generateSpecTE", "-noGenerateSpecTEBin", "-config", configName, "-teSpecOutDir", OUTPUT_PATH, mode },
+				new String[] { "-noGenerateSpecTEBin", "-config", configName, mode },
 				exitStatus);
 	}
 
 	protected FilenameToStream getResolver() {
-		return new SimpleFilenameToStream(OUTPUT_PATH);
+		return new SimpleFilenameToStream(getMetaDir());
 	}
 
 	@Override
@@ -68,6 +66,11 @@ public abstract class TraceExpressionSpecTest extends ModelCheckerTestCase {
 		return false;
 	}
 	
+	@Override
+	protected String getTESpecOutDir() {
+		return getMetaDir();
+	}
+
 	@Override
 	protected boolean doDumpTrace() {
 		return false;

--- a/tlatools/org.lamport.tlatools/test/tlc2/tool/liveness/ModelCheckerTestCase.java
+++ b/tlatools/org.lamport.tlatools/test/tlc2/tool/liveness/ModelCheckerTestCase.java
@@ -135,7 +135,7 @@ public abstract class ModelCheckerTestCase extends CommonTestCase {
 			final List<String> args = new ArrayList<String>(6);
 			
 			args.add("-metadir");
-			args.add(TLCGlobals.metaRoot + FileUtil.separator + getClass().getCanonicalName());
+			args.add(getMetaDir());
 			
 			// *Don't* check for deadlocks. All tests are interested in liveness
 			// checks which are shielded away by deadlock checking. TLC finds a
@@ -155,7 +155,7 @@ public abstract class ModelCheckerTestCase extends CommonTestCase {
 				// Make sure the generated spec ends up in a designated location.
 				args.add("-generateSpecTE");
 				args.add("-teSpecOutDir");
-				args.add(TTraceModelCheckerTestCase.getPath(getClass()));
+				args.add(getTESpecOutDir());
 			}
 			
 			if (noRandomFPandSeed()) {
@@ -213,7 +213,11 @@ public abstract class ModelCheckerTestCase extends CommonTestCase {
 			fail(e.getMessage());
 		}
 	}
-
+	
+	protected String getMetaDir() {
+		return TLCGlobals.metaRoot + FileUtil.separator + getClass().getCanonicalName();
+	}
+	
 	protected boolean noRandomFPandSeed() {
 		return true;
 	}
@@ -233,6 +237,10 @@ public abstract class ModelCheckerTestCase extends CommonTestCase {
 	protected boolean noGenerateSpec() {
 		return false;
 	}	
+	
+	protected String getTESpecOutDir() {
+		return TTraceModelCheckerTestCase.getPath(getClass());
+	}
 
 	protected void beforeTearDown() {
 		// No-op

